src/components/pages/UpdatePassword/index.tsx

```diff
@@ -1,8 +1,108 @@
 import * as React from 'react';
 import styled from '@emotion/styled';
+import { connect } from 'react-redux';
 
-export default function UpdatePassword(): React.ReactElement<{}> {
-  return <UpdatePassword.Wrapper>Update Password will go here.</UpdatePassword.Wrapper>;
+import Form from 'components/ui/Form';
+import withDashboardContainer, { DashboardProps } from 'components/containers/DashBoardContainer';
+import Input from 'components/ui/Input';
+import Checkbox from 'components/ui/Checkbox';
+import { changeLoggedInUserPassword } from 'store/actions/auth';
+import __spacing from 'settings/__spacing';
+
+interface UpdatePasswordProps {
+  handleSubmit: (trigger: string) => (values: Record<string, any>) => void;
+  callUpdatePassword: (payload: any) => void;
+  newPassword: string;
+  currentPassword: string;
+  confirmPassword: string;
+  status: {
+    hasError: boolean;
+    isLoading: boolean;
+  };
+}
+
+function UpdatePassword(props: UpdatePasswordProps): React.ReactElement<{}> {
+  const { handleSubmit, status } = props;
+  const defaults = {
+    newPassword: '',
+    currentPassword: '',
+    confirmPassword: '',
+  };
+  function isAnyInputBoxEmpty(composedProps) {
+    const { values } = composedProps;
+    const inputVals = [values.newPassword, values.currentPassword, values.confirmPassword];
+
+    return inputVals.some(value => value === '');
+  }
```

Comment:
This functionality has now been duplicated across features. Can we put it in the form so as to reuse it?

Reply:
Does not work in exactly the same way. Here the submit button is enabled when values have been inputted in all the text fields while in the `UpdateProfile` submit button is enabled when you input a new value(different from your current profile data) for _any_ of the fields. 

One more thing, I suggest that we actually repeat say 4 or more times before abstracting logic. That way we have enough examples and similarities of the different occurrences when abstracting

Reply:
Great.